samples/openthread/coap_client/src/coap_client.c

```diff
@@ -111,11 +187,39 @@ static void send_provisioning_request(struct k_work *item)
 {
 	ARG_UNUSED(item);
 
+	if (IS_ENABLED(CONFIG_OPENTHREAD_MTD_SED)) {
+		/* decrease the polling period for higher responsiveness */
+		if (false == poll_period_response_set()) {
+			LOG_WRN("Failed to set response poll period");
+			return;
+		}
+	}
+
 	LOG_INF("Send 'provisioning' request");
 	coap_send_request(COAP_METHOD_GET, &multicast_local_addr,
 			  provisioning_option, NULL, 0u, on_provisioning_reply);
 }
 
+static void toggle_minimal_sleepy_end_device(struct k_work *item)
+{
+	otError error;
+	struct otInstance *instance = openthread_get_default_instance();
+	otLinkModeConfig mode = otThreadGetLinkMode(instance);
+
+	if (mode.mRxOnWhenIdle) {
+		mode.mRxOnWhenIdle = false;
+		dk_set_led_off(MTD_SED_LED);
```

Comment:
I guess that `otThreadSetLinkMode` is not likely to fail, but shouldn't we actually modify the led only after the mode is successfully changed?

Reply:
Agree. Changed.